File: examples/python/src/003_weather.py
import asyncio
import httpx
from langchain_anthropic import ChatAnthropic
from langchain_core.tools import tool
from langchain.agents import create_agent

model = ChatAnthropic(model="claude-opus-4-8")


@tool
async def weather(latitude: float, longitude: float) -> dict:
    """Get the current weather for a given latitude and longitude"""
    async with httpx.AsyncClient() as client:
        response = await client.get(
            "https://api.open-meteo.com/v1/forecast",
            params={"latitude": latitude, "longitude": longitude, "current_weather": "true"},
        )
        return response.json()


agent = create_agent(model=model, tools=[weather])


async def main():
    response = await agent.ainvoke(
        {"messages": [{"role": "user", "content": "What is the weather in Zürich?"}]}
    )
    print(response["messages"][-1].content)


asyncio.run(main())

File: examples/js/src/003-weather.ts
import { tool } from "@langchain/core/tools";
import { ChatAnthropic } from "@langchain/anthropic";
import { createAgent } from "langchain";

const model = new ChatAnthropic({
  model: "claude-opus-4-8",
});

const weatherTool = tool(
  async ({ latitude, longitude }) => {
    const response = await fetch(
      `https://api.open-meteo.com/v1/forecast?latitude=${latitude}&longitude=${longitude}&current_weather=true`,
    );
    const data = await response.json();
    return data;
  },
  {
    name: "weather",
    description: "Get the current weather for a given latitude and longitude",
    schema: {
      type: "object",
      properties: {
        latitude: { type: "number", description: "Latitude of the location" },
        longitude: { type: "number", description: "Longitude of the location" },
      },
      required: ["latitude", "longitude"],
      additionalProperties: false,
    },
  },
);

const agent = createAgent({
  model: model,
  tools: [weatherTool],
});

const response = await agent.invoke({
  messages: [{ role: "user", content: "What is the weather in Mannheim?" }],
});

console.log(response.messages[response.messages.length - 1].content);
